sdk/helper/ldaputil/config.go

```diff
@@ -236,20 +262,31 @@ func NewConfigEntry(existing *ConfigEntry, d *framework.FieldData) (*ConfigEntry
 	if _, ok := d.Raw["certificate"]; ok || !hadExisting {
 		certificate := d.Get("certificate").(string)
 		if certificate != "" {
-			block, _ := pem.Decode([]byte(certificate))
-
-			if block == nil || block.Type != "CERTIFICATE" {
-				return nil, errors.New("failed to decode PEM block in the certificate")
-			}
-			_, err := x509.ParseCertificate(block.Bytes)
-			if err != nil {
-				return nil, errwrap.Wrapf("failed to parse certificate: {{err}}", err)
+			if err := validateCertificate([]byte(certificate)); err != nil {
+				return nil, errwrap.Wrapf("failed to parse server tls cert: {{err}}", err)
 			}
 		}
-
 		cfg.Certificate = certificate
 	}
 
+	if _, ok := d.Raw["client_tls_cert"]; ok || !hadExisting {
+		clientTLSCert := d.Get("client_tls_cert").(string)
+		cfg.ClientTLSCert = clientTLSCert
+	}
+
+	if _, ok := d.Raw["client_tls_key"]; ok || !hadExisting {
+		clientTLSKey := d.Get("client_tls_key").(string)
+		cfg.ClientTLSKey = clientTLSKey
+	}
+
+	if cfg.ClientTLSCert != "" && cfg.ClientTLSKey != "" {
+		if _, err := tls.X509KeyPair([]byte(cfg.ClientTLSCert), []byte(cfg.ClientTLSKey)); err != nil {
+			return nil, errwrap.Wrapf("failed to parse client X509 key pair: {{err}}", err)
+		}
+	} else if (cfg.ClientTLSCert != "" && cfg.ClientTLSKey == "") || (cfg.ClientTLSCert == "" && cfg.ClientTLSKey != "") {
```

Comment:
```suggestion
	} else if cfg.ClientTLSCert != "" || cfg.ClientTLSKey != "" {
```

Reply:
That hits the case where you don't use either of them (which happens in case you use other type of authentication). Do we really wanna throw an error in that case?